tests/meta/observations/test_observations_api.py

```diff
@@ -0,0 +1,529 @@
+    caplog.set_level(logging.DEBUG)
+
+    # GIVEN an observations API, a list of observation input files, and a workflow customer
+    observations_api: ObservationsAPI = request.getfixturevalue(
+        f"{workflow.replace('-', '_')}_observations_api"
+    )
+    observations_input_files: ObservationsInputFiles = request.getfixturevalue(
+        f"{workflow.replace('-', '_')}_observations_input_files"
+    )
+    customer: Customer = request.getfixturevalue(f"{workflow.replace('-', '_')}_customer")
+
+    # GIVEN a case eligible for Loqusdb uploads
+    case: Case = observations_api.store.get_case_by_internal_id(case_id)
+    case.customer.internal_id = customer.internal_id
+    case.samples[0].is_tumour = is_tumour
+
+    # GIVEN a mock scenario for a successful upload
+    mocker.patch.object(analysis_api, "get_data_analysis_type", return_value=sequencing_method)
+    mocker.patch.object(
+        ObservationsAPI, "get_observations_input_files", return_value=observations_input_files
+    )
+    mocker.patch.object(ObservationsAPI, "is_duplicate", return_value=False)
+    mocker.patch.object(LoqusdbAPI, "load", return_value={"variants": number_of_loaded_variants})
+    mocker.patch.object(
+        LoqusdbAPI, "get_case", return_value={"case_id": case_id, LOQUSDB_ID: loqusdb_id}
+    )
+    mocker.patch.object(LimsAPI, "get_source", return_value=SourceType.TISSUE)
+
+    # WHEN uploading the case observations to Loqusdb
+    observations_api.upload(case_id)
+
+    # THEN the case should be successfully uploaded
+    assert f"Uploaded {number_of_loaded_variants} variants to Loqusdb" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "workflow, analysis_api, sequencing_method, is_tumour",
+    [
+        (Workflow.BALSAMIC, BalsamicAnalysisAPI, CancerAnalysisType.TUMOR_WGS, True),
+        (Workflow.MIP_DNA, MipDNAAnalysisAPI, SequencingMethod.WGS, False),
+    ],
+)
+def test_observations_upload_not_eligible(
+    case_id: str,
+    loqusdb_id: str,
+    number_of_loaded_variants: int,
+    workflow: Workflow,
+    analysis_api: AnalysisAPI,
+    sequencing_method: str,
+    is_tumour: bool,
+    request: FixtureRequest,
+    mocker: MockFixture,
+    caplog: LogCaptureFixture,
+):
+    """Test upload of observations."""
+    caplog.set_level(logging.DEBUG)
+
+    # GIVEN an observations API, a list of observation input files, and a workflow customer
+    observations_api: ObservationsAPI = request.getfixturevalue(
+        f"{workflow.replace('-', '_')}_observations_api"
+    )
+    observations_input_files: ObservationsInputFiles = request.getfixturevalue(
+        f"{workflow.replace('-', '_')}_observations_input_files"
+    )
+    customer: Customer = request.getfixturevalue(f"{workflow.replace('-', '_')}_customer")
+
+    # GIVEN a case eligible for Loqusdb uploads
+    case: Case = observations_api.store.get_case_by_internal_id(case_id)
+    case.customer.internal_id = customer.internal_id
+    case.samples[0].is_tumour = is_tumour
```

Comment:
to avoid code repetition warning: This case could be a fixture

Reply:
Moved some parts to the observations fixture and collapsed the duplicated code using parametrisation 🥵